src/components/partnership/partnership.service.ts

```diff
@@ -603,4 +628,67 @@ export class PartnershipService {
       );
     }
   }
+
+  protected async isFirstPartnership(projectId: string): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .match([
+        node('project', 'Project', { id: projectId }),
+        relation('out', '', 'partnership', { active: true }),
+        node('partnership'),
+      ])
+      .return(['partnership'])
+      .asResult<{ partnership?: Node }>()
+      .first();
+
+    return result?.partnership ? false : true;
+  }
+
+  /**
+   *
+   * match current primary partnership, its project, and all other partnerships
+   * set current to primary false
+   * set first of other partnerships to primary true
+   */
+  protected async movePrimaryPartnership(
+    direction: 'to' | 'from',
+    partnershipId: string
+  ): Promise<void> {
+    const createdAt = DateTime.local();
+    const newPropertyNodeProps = {
+      createdAt,
+      value: direction === 'from',
+      sortValue: determineSortValue(direction === 'from'),
+    };
+    const query = this.db
+      .query()
+      .match([
+        node('partnership', 'Partnership', { id: partnershipId }),
+        relation('in', '', 'partnership', { active: true }),
+        node('project', 'Project'),
+        relation('out', '', 'partnership', { active: true }),
+        node('otherPartnership'),
+      ])
+      .with('otherPartnership')
+      .limit(1)
+      .optionalMatch([
+        node('otherPartnership'),
+        relation('out', 'oldRel', 'primary', { active: true }),
+        node('property', 'Property'),
+      ])
+      .setValues({
+        'oldRel.active': false,
+      })
```

Comment:
If our direction is `to` then shouldn't this apply for all `otherPartnership` not just `limit 1`?

Reply:
Wait we need to match the other one whose property is changing, and then create a property there as well with the opposite value.